src/topics/group-indexer/group-indexer.ts

```diff
@@ -0,0 +1,54 @@
+import { GroupIndexerServiceConstructorArgs } from "./group-indexer.types";
+import { LoggerService } from "logger";
+import { GroupSnapshotStore } from "topics/group-snapshot";
+import { GroupStore } from "topics/group/group.store";
+
+type AccountMap = {
+  [accountIdentifier: string]: Set<string>;
+};
+
+export class GroupIndexerService {
+  groupStore: GroupStore;
+  groupSnapshotStore: GroupSnapshotStore;
+  logger: LoggerService;
+
+  constructor({
+    groupStore,
+    groupSnapshotStore,
+    logger,
+  }: GroupIndexerServiceConstructorArgs) {
+    this.groupStore = groupStore;
+    this.groupSnapshotStore = groupSnapshotStore;
+    this.logger = logger;
+  }
+
+  public async indexGroups(): Promise<void> {
+    const map: AccountMap = {};
+
+    const groups = await this.groupStore.all();
+    
+    for (const groupName in groups) {
+      const group = groups[groupName];
+      const groupId = group.id;
+
+      // TODO: Pick the latest snapshot in registry-tree
+      const snapshot = await group.resolvedIdentifierData();
```

Comment:
The latest snapshot shoule be retrieved from the groupSnapshotStore thanks to the groupId